llvm/lib/ProfileData/SampleProfReader.cpp

```diff
@@ -902,12 +890,16 @@ std::error_code SampleProfileReaderExtBinaryBase::readFuncProfiles() {
       for (const auto &NameOffset : FuncOffsetList) {
         const auto &FContext = NameOffset.first;
         auto FName = FContext.getName();
+        StringRef FNameString;
+        if (!useMD5()) {
+          FNameString = FName.stringRef(Buffer);
+        }
         // For function in the current module, keep its farthest ancestor
         // context. This can be used to load itself and its child and
         // sibling contexts.
-        if ((useMD5() && FuncGuidsToUse.count(std::stoull(FName.data()))) ||
-            (!useMD5() && (FuncsToUse.count(FName) ||
-                           (Remapper && Remapper->exist(FName))))) {
+        if ((useMD5() && FuncGuidsToUse.count(FName.getHashCode())) ||
+            (!useMD5() && (FuncsToUse.count(FNameString) ||
```

Comment:
does it cost more compile time to convert to string when MD5 is not used?

Reply:
If MD5 is not used, ProfileFuncRef should be a StringRef and the convert function should be zero cost.

Reply:
Actually regarding this, because @MatzeB mentioned the interface "StringRef ProfileFuncRef::stringRef(std::string &Buffer)" is not elegant, would it be better to just have "StringRef ProfileFuncRef::stringRef()" and only allow it being used if we know the ProfileFuncRef is representing a string for sure, and let the compiler crash if used otherwise?

Reply:
yes -- having a std::string buffer which is not used here is very confusing.